app/Platform/Listeners/DispatchUpdatePlayerStatsJob.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Platform\Listeners;
+
+use App\Platform\Events\PlayerBadgeAwarded;
+use App\Platform\Events\PlayerBadgeLost;
+use App\Platform\Events\PlayerRankedStatusChanged;
+use App\Platform\Jobs\UpdatePlayerStatsJob;
+use App\Site\Models\User;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class DispatchUpdatePlayerStatsJob implements ShouldQueue
+{
+    public function handle(object $event): void
+    {
+        $user = null;
+
+        switch ($event::class) {
+            case PlayerBadgeAwarded::class:
+            case PlayerRankedStatusChanged::class:
+                $user = $event->playerBadge->user;
```

Comment:
This feels incorrect. `PlayerRankedStatusChanged` does not have a `playerBadge` member. I think you meant to group it with the `PlayerBadgeLost` handler.

Reply:
Yep, silly typo. Should be all set now.